Sources/TuistKit/Commands/GenerateCommand.swift

```diff
@@ -50,6 +55,7 @@ class GenerateCommand: NSObject, Command {
         let path = self.path(arguments: arguments)
         let projectOnly = arguments.get(projectOnlyArgument) ?? false
 
+        try signingInstaller.installSigning(at: path)
```

Comment:
This approach might result in the following:
- Installing certs and profiles that are not needed by the project being generated.
- Installing certs and profiles that are not valid for the project.

We'd end up with the same problems that developers are currently experiencing when dealing with certificates: I placed things in my system and I expect things to work but they don't. *What am I doing wrong?*.

For that reason, I suggest taking a different approach here. We need to make sure we **install what's needed** and that **the project and the environment is ready** to sign. That means that all this logic needs to be part of the generation of the project, and not a step before or after it. I suggest leveraging [GraphMappers](https://github.com/tuist/tuist/blob/master/Sources/TuistCore/Graph/Mappers/GraphMapper.swift#L5):

```swift
class ConfigureSigningGraphMapper: GraphMapping {
  func map(graph: Graph) throws -> (Graph, [SideEffectDescriptor]) {
      // 1. Get the list of certificates and profiles (IO)
      // 2. Iterate through all the targets in the project.
      // 3. If a target has signing artifacts (profile and certificate) adjust the build settings of the target accordingly.
      // 4. Return a graph with the targets containing the right build settings. And the following side effects:
      //     .installProfile(path: "profilepath")
      //     .installCertificate(path: "certificatepath")
      //.    .validateCertificate(path: "certificatepath")
      //.    .validateProfile(path: "profilepath)
      //.    .validateProfileAndCertificateMatch(profile: "xx", certificate: "xx")
  }
}
```